RealmSwift/Realm.swift

```diff
@@ -366,6 +369,112 @@ import Realm.Private
         return rlmRealm.inWriteTransaction
     }
 
+// MARK: Asynchronous Transactions
+
+    /**
+     Asynchronously performs actions contained within the given block inside a write transaction.
+     The write transaction is begun asynchronously as if calling `beginAsyncWrite`,
+     and by default the transaction is commited asynchronously after the block completes.
+     You can also explicitly call `commitWrite` or `cancelWrite` from
+     within the block to synchronously commit or cancel the write transaction.
+
+     @param block The block containing actions to perform.
+
+     @param completionBlock A block which will be called on the source thread or queue
+                        once the commit has either completed or failed with an error.
+
+     @return An id identifying the asynchronous transaction which can be passed to
+             `cancelAsyncWrite` prior to the block being called to cancel
+             the pending invocation of the block.
+    */
+    @discardableResult
+    public func writeAsync(_ block: @escaping (AsyncTransactionId) -> Void, _ onComplete: ((Swift.Error?) -> Void)? = nil) -> AsyncTransactionId {
```

Comment:
Completion callback should have an argument label.

Why does this and `beginAsyncWrite()` pass the transaction Id to the callback? The id is no longer usable for anything once the callback is invoked.

Reply:
Transaction Id is not required in `beginAsyncWrite` any more and can be removed. Thanks